clients/cli/src/main.rs

```diff
@@ -0,0 +1,552 @@
+
+    let ignore_filter = |report: &ViolationReport| {
+        command_config
+            .epoch
+            .map(|e| e != report.epoch())
+            .unwrap_or_default()
+            || node_pubkey
+                .map(|pk| pk != report.pubkey)
+                .unwrap_or_default()
+            || reporter_pubkey
+                .map(|pk| pk != report.reporter)
+                .unwrap_or_default()
+    };
+
+    let gpa_config = RpcProgramAccountsConfig {
+        account_config: RpcAccountInfoConfig {
+            encoding: Some(UiAccountEncoding::Base64),
+            commitment: Some(config.rpc_client.commitment()),
+            ..RpcAccountInfoConfig::default()
+        },
+        ..RpcProgramAccountsConfig::default()
+    };
+
+    let reports = config
+        .rpc_client
+        .get_program_accounts_with_config(&spl_slashing::id(), gpa_config)
+        .await?;
+
+    let mut displays = vec![];
+    for (pubkey, report_account) in reports {
+        let report: &ViolationReport =
+            pod_from_bytes(&report_account.data()[..std::mem::size_of::<ViolationReport>()])?;
+        if ignore_filter(report) {
+            continue;
+        }
+        let proof = report_account.data()[std::mem::size_of::<ViolationReport>()..].to_vec();
+        displays.push(ViolationReportOutput::from_report(pubkey, report, proof));
+    }
+
+    displays.sort_by(|a, b| a.slot.cmp(&b.slot));
+
+    Ok(format_output(
+        config,
+        "Display".to_string(),
+        ViolationReportListOutput(displays),
+    ))
+}
+
+async fn initialize_and_write_proof(
+    config: &Config,
+    payer: &Arc<dyn Signer>,
+    proof_account: &Keypair,
+    proof_type: ProofType,
+    proof_data: &[u8],
+) -> Result<(), Error> {
+    // Initialize account with the record program
+    let account_length = proof_type
+        .proof_account_length()
+        .saturating_add(pod_get_packed_len::<RecordData>());
+    let lamports = 1.max(config.rent.minimum_balance(account_length));
+    let signers: [&dyn Signer; 2] = [&**payer, proof_account];
+    let initialize_ix =
+        spl_record::instruction::initialize(&proof_account.pubkey(), &payer.pubkey());
+    let transaction = Transaction::new_signed_with_payer(
+        &[
+            solana_system_interface::instruction::create_account(
+                &payer.pubkey(),
+                &proof_account.pubkey(),
+                lamports,
+                account_length as u64,
+                &spl_record::id(),
+            ),
+            initialize_ix,
+        ],
+        Some(&payer.pubkey()),
+        &signers,
+        config.program_client.get_latest_blockhash().await?,
+    );
+    if process_transaction(config, transaction).await?.is_some() {
+        println_display(
+            config,
+            format!("Initialized proof account {}", proof_account.pubkey()),
+        )
+    }
+
+    // Write the proof
+    let mut offset = 0;
+    let chunk_size = 800;
```

Comment:
I'm guessing this is just some good-enough heuristic for how much proof data can fit in a tx?

Reply:
yeah maybe could actually derive the overhead. This results in 4 transactions to write the proof as it's a little over 2400 bytes

Reply:
got it https://github.com/solana-program/slashing/issues/37